simtools/production_configuration/calculate_statistical_errors_grid_point.py

```diff
@@ -0,0 +1,472 @@
+"""
+Provides functionality to evaluate statistical errors from FITS files.
+
+Classes
+-------
+StatisticalErrorEvaluator
+    Handles error calculation for given FITS files and specified metrics.
+
+
+"""
+
+import logging
+
+import numpy as np
+from astropy.io import fits
+
+
+class StatisticalErrorEvaluator:
+    """
+    Evaluates statistical errors from a FITS file.
+
+    Parameters
+    ----------
+    file_path : str
+        Path to the FITS file.
+    file_type : str
+        Type of the file, either 'On-source' or 'Offset'.
+    metrics : dict, optional
+        Dictionary of metrics to evaluate. Default is None.
+    grid_point : tuple, optional
+        Tuple specifying the grid point (energy, azimuth, zenith, NSB, offset).
+    """
+
+    def __init__(
+        self,
+        file_path: str,
+        file_type: str,
+        metrics: dict[str, float] | None = None,
+        grid_point: tuple[float, float, float, float, float] | None = None,
+    ):
+        """
+        Initialize the evaluator with a specific FITS file, its type, and metrics to calculate.
+
+        Parameters
+        ----------
+        file_path : str
+            The path to the FITS file.
+        file_type : str
+            The type of the file ('On-source' or 'Offset').
+        metrics : dict, optional
+            Dictionary specifying which metrics to compute and their reference values.
+        grid_point : tuple, optional
+            Tuple specifying the grid point (energy, azimuth, zenith, NSB, offset).
+        """
+        self.file_path = file_path
+        self.file_type = file_type
+        self.metrics = metrics or {}
+        self.grid_point = grid_point
+
+        self.data = self.load_data_from_file()
+
+        self.error_eff_area = None
+        self.error_sig_eff_gh = None
+        self.error_energy_estimate_bdt_reg_tree = None
+        self.sigma_energy = None
+        self.delta_energy = None
+        self.error_gamma_ray_psf = None
+        self.error_image_template_methods = None
+
+        self.metric_results = None
+        self.scaled_events = None
+        self.scaled_events_gridpoint = None
+        self.energy_threshold = None
+
+    def load_data_from_file(self):
+        """
+        Load data from the FITS file and return relevant arrays.
+
+        Returns
+        -------
+        dict
+            Dictionary containing arrays from the FITS file.
+        """
+        data = {}
+        try:
+            with fits.open(self.file_path) as hdul:
+                events_data = hdul["EVENTS"].data  # pylint: disable=E1101
+                event_energies = events_data["ENERGY"]
+                mc_energies = events_data["MC_ENERGY"]
+
+                sim_events_data = hdul["SIMULATED EVENTS"].data  # pylint: disable=E1101
+                bin_edges_low = sim_events_data["MC_ENERG_LO"]
+                bin_edges_high = sim_events_data["MC_ENERG_HI"]
```

Comment:
Is the assumption that we want to pass the metric over the energy range given by the DL2 file good? Would it make sense to define a `valid energy range` for a given metric? e.g. generate a simulation production configuration with a 0.1% statistical uncertainty in the 30 GeV to 300 TeV range (although we simulated from 10 GeV to 500 TeV?)

Reply:
Already added a validity range with units to each metric in the yaml files.